src/PathContainer.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Dhii\Container;
+
+use Psr\Container\ContainerInterface;
+use function array_filter;
+use function ltrim;
+
+/**
+ * A container implementation that wraps around another to provide access to its values via partial path-like keys.
+ *
+ * @since [*next-version*]
+ */
+class PathContainer implements ContainerInterface
+{
+    /**
+     * @var ContainerInterface
+     */
+    protected $inner;
+
+    /**
+     * @var string
+     */
+    protected $root;
+
+    /**
+     * @var string
+     */
+    protected $delimiter;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerInterface $inner The container to decorate.
+     * @param string             $delimiter
+     */
+    public function __construct($inner, $delimiter = '/')
+    {
+        $this->inner = $inner;
+        $this->root = '';
+        $this->delimiter = $delimiter;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @since [*next-version*]
+     */
+    public function get($key)
+    {
+        $tKey = ltrim($key, $this->delimiter);
```

Comment:
The delimiter can be more than one key. In this case, looks like the result would be not what is expected:

`ltrim($key = '-key', $delimiter = '->') // 'key'`

Reply:
True! Looks like the trim functions won't be good here.